docs/client_middleware_cookbook.rst

```diff
@@ -150,6 +150,9 @@ A retry middleware that automatically retries failed requests with exponential b
                     )
                     return response
 
+                # Consume response body before retrying
+                await response.discard_content()
```

Comment:
We can't do it here, as this middleware knows nothing about the request. If it's a streaming response, this is just an infinite loop.

Reply:
Not really sure how to handle this case